src/test/storage/table_test.cpp

```diff
@@ -12,63 +12,81 @@
-//   EXPECT_EQ(t.column_name(ColumnID{1}), "col_2");
-//   // TODO(anyone): Do we want checks here?
-//   // EXPECT_THROW(t.column_name(ColumnID{2}), std::exception);
-// }
-
-// TEST_F(StorageTableTest, GetColumnType) {
-//   EXPECT_EQ(t.column_type(ColumnID{0}), "int");
-//   EXPECT_EQ(t.column_type(ColumnID{1}), "string");
-//   // TODO(anyone): Do we want checks here?
-//   // EXPECT_THROW(t.column_type(ColumnID{2}), std::exception);
-// }
-
-// TEST_F(StorageTableTest, GetColumnIdByName) {
-//   EXPECT_EQ(t.column_id_by_name("col_2"), 1u);
-//   EXPECT_THROW(t.column_id_by_name("no_column_name"), std::exception);
-// }
-
-// TEST_F(StorageTableTest, GetChunkSize) { EXPECT_EQ(t.target_chunk_size(), 2u); }
+class StorageTableTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    t.add_column("col_1", "int");
+    t.add_column("col_2", "string");
+  }
+
+  Table t{2};
+};
+
+TEST_F(StorageTableTest, ChunkCount) {
+  EXPECT_EQ(t.chunk_count(), 1u);
+  t.append({4, "Hello,"});
+  t.append({6, "world"});
+  t.append({3, "!"});
+  EXPECT_EQ(t.chunk_count(), 2u);
+}
+
+TEST_F(StorageTableTest, GetChunk) {
+  t.get_chunk(ChunkID{0});
+  // TODO(anyone): Do we want checks here?
+  // EXPECT_THROW(t.get_chunk(ChunkID{q}), std::exception);
+  t.append({4, "Hello,"});
+  t.append({6, "world"});
+  t.append({3, "!"});
+  t.get_chunk(ChunkID{1});
+}
+
+TEST_F(StorageTableTest, ColumnCount) { EXPECT_EQ(t.column_count(), 2u); }
+
+TEST_F(StorageTableTest, RowCount) {
+  EXPECT_EQ(t.row_count(), 0u);
+  t.append({4, "Hello,"});
+  t.append({6, "world"});
+  t.append({3, "!"});
+  EXPECT_EQ(t.row_count(), 3u);
+}
+
+TEST_F(StorageTableTest, GetColumnName) {
+  EXPECT_EQ(t.column_name(ColumnID{0}), "col_1");
+  EXPECT_EQ(t.column_name(ColumnID{1}), "col_2");
+  // TODO(anyone): Do we want checks here?
+  // EXPECT_THROW(t.column_name(ColumnID{2}), std::exception);
+}
+
+TEST_F(StorageTableTest, GetColumnType) {
+  EXPECT_EQ(t.column_type(ColumnID{0}), "int");
+  EXPECT_EQ(t.column_type(ColumnID{1}), "string");
+  // TODO(anyone): Do we want checks here?
+  // EXPECT_THROW(t.column_type(ColumnID{2}), std::exception);
+}
+
+TEST_F(StorageTableTest, GetColumnIdByName) {
+  EXPECT_EQ(t.column_id_by_name("col_2"), 1u);
+  EXPECT_THROW(t.column_id_by_name("no_column_name"), std::exception);
+}
+
+TEST_F(StorageTableTest, GetChunkSize) { EXPECT_EQ(t.target_chunk_size(), 2u); }
+
+TEST_F(StorageTableTest, AddColumn) {
+  t.add_column("col_3", "string");
+  EXPECT_EQ(t.column_count(), 3u);
+}
+
+TEST_F(StorageTableTest, AddColumnFail) {
+  t.append({42, "Space Traveler"});
+  EXPECT_THROW(t.add_column("col_3", "string"), std::exception);
+}
+
+TEST_F(StorageTableTest, PrivateAddSegment) {
+  t.add_column("col_3", "int");
+  auto& currentChunk = t.get_chunk(ChunkID{0});
+  auto lastSegment = currentChunk.get_segment(ColumnID{2});
+  EXPECT_EQ(currentChunk.column_count(), 3u);
+  EXPECT_THROW(lastSegment->append("String"), std::exception);
+}
```

Comment:
Is this supposed to test that `String` cannot be added to a segment of type `int`? Then, this test should be named differently.